src/test/java/spoon/test/template/SubstitutionTest.java

```diff
@@ -47,4 +47,36 @@ public void testSubstitutionInsertAllFields() {
         @Override
         public void statement() { }
     }
+
+    @Test
+    public void testInsertAllNestedTypes() {
+        // contract: Substitution.insertAllNestedTypes inserts the only nested class from a singly nested template into the target class
+
+        // arrange
+        Launcher spoon = new Launcher();
+        spoon.addTemplateResource(new FileSystemFile("./src/test/java/spoon/test/template/SubstitutionTest.java"));
+
+        spoon.buildModel();
+        Factory factory = spoon.getFactory();
+
+        CtType<?> targetType = factory.Class().create("goodClass");
+        StatementTemplate template = new SinglyNestedTemplate();
+
+        // act
+        Substitution.insertAllNestedTypes(targetType, template);
+
+        // assert
+        assertEquals(1, targetType.getNestedTypes().size());
+        assertEquals("nestedClass", targetType.getNestedType("nestedClass").getSimpleName());
```

Comment:
The new assertion strengthens the test case in the sense that it verifies that there's a nested type with name `nestedClass`, and that's what was requested. So I think this is fine now.

However, I just wanted to note that the layout of the assertion makes it look like a test for `CtType.getNestedType(String)`. To be clear, it looks like the assertion verifies that getting a nested type by a given name returns a type with that name. It would be sufficient here to just verify that the returned type is non-null, due to the functionality of `getNestedType`.

```suggestion
        assertNotNull(targetType.getNestedType("nestedClass"));
```

Personally, I would probably have written both assertions like this instead, so as not to have to query `targetType` multiple times.

```java
        List<CtType<?>> nestedTypes = targetType.getNestedTypes();
        assertThat(nestedTypes.size(), equalTo(1));
        assertThat(nestedTypes.get(0), equalTo("nestedClass"));
```

Reply:
Thanks for the suggestions, so I can add this in the PR that I will raise for another new test?